commit 9e1881da167aa4db0d59bb4bfb63aaa330517b11
Author:     Isaac Yang <isaacy@nvidia.com>
AuthorDate: Wed Mar 22 13:24:28 2023 -0700
Commit:     Isaac Yang <isaacy@nvidia.com>
CommitDate: Wed Mar 22 14:32:13 2023 -0700

    Fix incorrect server status after job aborted and server restarted
---
 nvflare/private/fed/server/job_runner.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/nvflare/private/fed/server/job_runner.py b/nvflare/private/fed/server/job_runner.py
index 379144af..bd6a91ce 100644
--- a/nvflare/private/fed/server/job_runner.py
+++ b/nvflare/private/fed/server/job_runner.py
@@ -63,7 +63,7 @@ class JobRunner(FLComponent):
         if event_type == EventType.SYSTEM_START:
             engine = fl_ctx.get_engine()
             self.scheduler = engine.get_component(SystemComponents.JOB_SCHEDULER)
-        elif event_type in [EventType.JOB_COMPLETED, EventType.JOB_ABORTED, EventType.JOB_CANCELLED]:
+        elif event_type in [EventType.JOB_COMPLETED, EventType.END_RUN]:
             self._save_workspace(fl_ctx)
         elif event_type == EventType.SYSTEM_END:
             self.stop()
